### PR TITLE
Exporting KUBE_MASTER_OS_DISTRIBUTION for 1.3+ and KUBE_OS_DISTRIBUTION for 1.2 for GCI tests.

### PR DESCRIPTION
**Pull request by maisem**
@wonderfly @spxtr

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml [hunk 1]
```diff
@@ -88,6 +88,7 @@
                 export GINKGO_TEST_ARGS="--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]"
                 export GINKGO_PARALLEL="y"
                 export PROJECT="e2e-gce-gci-ci-master"
+                export KUBE_MASTER_OS_DISTRIBUTION="gci"
```
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml]: I believe `KUBE_NODES_OS_DISTRIBUTION` defaults to `KUBE_MASTER_OS_DISTRIBUTION` so we don't have to specify it here?

Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml]: Also, can you move this up to the `provider-env` in the job-template? That way you only have to do it once.

**Comment by wonderfly** [comment]
Thanks for making the change.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml (hunk 1 above)
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml]: Don't seem so. We may want to set `KUBE_NODE_OS_DISTRIBUTION` to `gci` too.

Comment (maisem) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml]: Sorry, I thought we were only testing GCI masters. Added KUBE_NODE_OS_DISTRIBUTION as well.

**Comment by spxtr** [comment]
YAML looks fine to me.

**Comment by wonderfly** [comment]
LGTM

**Pull request merged**